virtual_ecosystem/models/litter/env_factors.py

```diff
@@ -5,6 +5,81 @@
 
 import numpy as np
 from numpy.typing import NDArray
+from xarray import DataArray
+
+from virtual_ecosystem.core.core_components import LayerStructure
+from virtual_ecosystem.models.litter.constants import LitterConsts
+
+
+def calculate_environmental_factors(
+    air_temperatures: DataArray,
+    soil_temperatures: DataArray,
+    water_potentials: DataArray,
+    layer_structure: LayerStructure,
+    constants: LitterConsts,
+):
+    """Calculate the impact of the environment has on litter decay across litter layers.
+
+    For the above ground layer the impact of temperature is calculated, and for the
+    below ground layer the effect of temperature and soil water potential are both
+    calculated.
+
+    The relevant above ground temperature is the surface temperature, which can be
+    easily extracted from the temperature data. It's more complex for the below ground
+    temperature and the water potential as the relevant values are averages across the
+    microbially active depth. These are calculated by averaging across the soil layers
+    with each layer weighted by the proportion of the total microbially active depth it
+    represents.
+
+    Args:
+        air_temperatures: Air temperatures, for all above ground layers [C]
+        soil_temperatures: Soil temperatures, for all soil layers [C]
+        water_potentials: Water potentials, for all soil layers [kPa]
+        layer_structure: The LayerStructure instance for the simulation.
+        constants: Set of constants for the litter model
+
+    Returns:
+        A dictionary containing three environmental factors, one for the effect of
+        temperature on above ground litter decay, one for the effect of temperature on
+        below ground litter decay, and one for the effect of soil water potential on
+        below ground litter decay.
+    """
+
+    surface_temp = air_temperatures[layer_structure.index_surface_scalar].to_numpy()
+    below_ground_temp = average_over_microbially_active_layers(
+        environmental_variable=soil_temperatures, layer_structure=layer_structure
+    )
+    water_potential = average_over_microbially_active_layers(
+        environmental_variable=water_potentials, layer_structure=layer_structure
+    )
+
+    # Calculate temperature factor for the above ground litter layers
+    temperature_factor_above = calculate_temperature_effect_on_litter_decomp(
```

Comment:
you could probably combine above and below ground te,temperature calculation to make it slightly more concise, but I think it's also ok like this for readability

Reply:
combine as in wrap them up into a single function, or like a list comprehension?

Reply:
I thought of list comprehension (do write the same thing twice)

Reply:
Ahh gotcha! List compressions still confuse/scare me, but that something I probably should get over 😅

Reply:
I think it is good to do it in two steps, write it all out first and then when you realize that you've written the same function several times it's quite straight forward to loop over the variables :-)